nautobot/extras/tests/test_jobs.py

```diff
@@ -606,6 +606,32 @@ def test_job_profiling(self):
         self.assertTrue(profiling_result.exists())
         profiling_result.unlink()
 
+    def test_job_singleton(self):
+        module = "singleton"
+        name = "TestSingletonJob"
+
+        job_class, _ = get_job_class_and_model(module, name, "local")
+        self.assertTrue(job_class.is_singleton)
+        cache.set(job_class.singleton_cache_key, 1)
+        failed_job_result = create_job_result_and_run_job(module, name)
+
+        self.assertEqual(
+            failed_job_result.status, JobResultStatusChoices.STATUS_FAILURE, msg="Duplicate singleton job didn't error."
+        )
+
+    def test_job_ignore_singleton(self):
+        module = "singleton"
+        name = "TestSingletonJob"
+
+        job_class, _ = get_job_class_and_model(module, name, "local")
+        self.assertTrue(job_class.is_singleton)
+        cache.set(job_class.singleton_cache_key, 1)
+        failed_job_result = create_job_result_and_run_job(module, name, _ignore_singleton_lock=True)
```

Comment:
Would be good to also enhance the `test_rerun_job` test in `test_views.py` to make sure that a job rerun correctly inherits the `_ignore_singleton_lock` setting from the original job run (i.e. the code you added below in `views.py`).

Reply:
I will tackle this once the other conversation is resolved.

Reply:
Done!